console/console-init/ui/src/modules/address-space/containers/AddressSpaceListContainer/AddressSpaceListContainer.tsx

```diff
@@ -0,0 +1,245 @@
+    dispatch({
+      type: types.SHOW_MODAL,
+      modalType: MODAL_TYPES.EDIT_ADDRESS_SPACE,
+      modalProps: {
+        addressSpace
+      }
+    });
+  };
+
+  const onDeleteAddressSpace = (addressSpace: IAddressSpace) => {
+    if (addressSpace) {
+      const variables = {
+        a: {
+          name: addressSpace.name,
+          namespace: addressSpace.nameSpace
+        }
+      };
+      setDeleteAddressSpaceQueryVariables(variables);
+    }
+  };
+
+  const onChangeDelete = (addressSpace: IAddressSpace) => {
+    dispatch({
+      type: types.SHOW_MODAL,
+      modalType: MODAL_TYPES.DELETE_ADDRESS_SPACE,
+      modalProps: {
+        selectedItems: [addressSpace.name],
+        data: addressSpace,
+        onConfirm: onDeleteAddressSpace,
+        option: "Delete",
+        detail: `Are you sure you want to delete this addressspace: ${addressSpace.name} ?`,
+        header: "Delete this Address Space ?"
+      }
+    });
+  };
+
+  //Download the certificate function
+  const onDownloadCertificate = async (data: IObjectMeta_v1_Input) => {
+    const dataToDownload = await client.query({
+      query: DOWNLOAD_CERTIFICATE,
+      variables: {
+        as: {
+          name: data.name,
+          namespace: data.namespace
+        }
+      },
+      fetchPolicy: FetchPolicy.NETWORK_ONLY
+    });
+    if (dataToDownload.errors) {
+      console.log("Error while download", dataToDownload.errors);
+    }
+    const url = window.URL.createObjectURL(
+      new Blob([dataToDownload.data.messagingCertificateChain])
+    );
+    const link = document.createElement("a");
+    link.href = url;
+    link.setAttribute("download", `${data.name}.crt`);
+    document.body.appendChild(link);
+    link.click();
+    if (link.parentNode) link.parentNode.removeChild(link);
+  };
+
+  const addressSpacesList: IAddressSpace[] = addressSpaces.addressSpaces.map(
+    addSpace => ({
+      name: addSpace.metadata.name,
+      nameSpace: addSpace.metadata.namespace,
+      creationTimestamp: addSpace.metadata.creationTimestamp,
+      type: addSpace.spec.type,
+      planValue: addSpace.spec.plan.metadata.name,
+      displayName: addSpace.spec.plan.spec.displayName,
+      isReady: addSpace.status && addSpace.status.isReady,
+      phase:
+        addSpace.status && addSpace.status.phase ? addSpace.status.phase : "",
+      messages:
+        addSpace.status && addSpace.status.messages
+          ? addSpace.status.messages
+          : [],
+      authenticationService:
+        addSpace.spec &&
+        addSpace.spec.authenticationService &&
+        addSpace.spec.authenticationService.name,
+      selected:
+        selectedAddressSpaces.filter(({ name, nameSpace }) =>
+          compareObject(
+            { name, nameSpace: addSpace.metadata.namespace },
+            { name: addSpace.metadata.name, nameSpace }
+          )
+        ).length === 1
+    })
+  );
+
+  const onSort = (_event: any, index: any, direction: any) => {
```

Comment:
The type needs to be fixed here as well.

Reply:
this is fixed